astropy/tests/tests/test_quantity_helpers.py

```diff
@@ -0,0 +1,35 @@
+from ... import units as u
+
+from ..helper import assert_quantity_allclose, pytest
+
+
+def test_assert_quantity_allclose():
+
+    assert_quantity_allclose([1,2], [1,2])
+
+    assert_quantity_allclose([1,2] * u.m, [100,200] * u.cm)
+
+    assert_quantity_allclose([1,2] * u.m, [101,201] * u.cm, atol=2 * u.cm)
+
+    with pytest.raises(AssertionError):
+        assert_quantity_allclose([1,2] * u.m, [90,200] * u.cm)
+
+    with pytest.raises(AssertionError):
+        assert_quantity_allclose([1,2] * u.m, [101,201] * u.cm, atol=0.5 * u.cm)
+
+    with pytest.raises(TypeError) as exc:
```

Comment:
``UnitsError`` will get raised if the units are wrong, but if the user passes a numpy array or python float then it is a ``TypeError`` because it's not the right kind of object.